### PR TITLE
Update the README concerning write modes.

### PR DESCRIPTION
**Pull request by markstory**
Add checkstyle and more detail on each of the write modes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,14 +51,17 @@ read data from stdin. Alternatively, you can use `cargo fmt` to format all
...
-rustfmt does not backup the source files. To print the output to stdout, use the
-display mode. The write mode can be set by passing the `--write-mode` flag on
-the command line.
-
-`rustfmt filename --write-mode=display` prints the output of rustfmt to the
-screen, for example.
+diff, replace, overwrite, display, coverage, and checkstyle.
+
+* `replace` Is the default and overwrites the original files after renaming them.
+* `overwrite` rustfmt does not backup the source files.
```
Comment (nrc) [inline README.md]: Nit: capital R. Probably better to say that `replace` does make backups, and overwrite overwrites the original file (without backup).

File: README.md [hunk 2]
```diff
@@ -51,14 +51,17 @@ read data from stdin. Alternatively, you can use `cargo fmt` to format all
...
+* `display` Will print the formatted file to stdout.
+* `diff` Will print a diff between the current file and formatted file to stdout.
```
Comment (nrc) [inline README.md]: What does "current" mean? The one specified on the command line?

**Comment by nrc** [comment]
Looks good, thank you!

**Comment by markstory** [comment]
@nrc I think I've addressed the feedback you had.

**Comment by nrc** [comment]
Cool, thanks!

**Pull request merged**